### PR TITLE
Add continuous integration workflow

### PR DESCRIPTION
**Pull request by cassava**
This is a simple Github Actions workflow that will run a cloe build + smoketest on each push.
If we find this is too much, we can also limit it to pull-requests only.

### CODE REVIEW

**Comment by cassava** [comment]
It currently takes about 20 minutes to run this job. Kinda longer than I was hoping, but there is still a lot of room for improvement. So far I have not added any kind of caching optimization, for example.

See: https://github.com/marketplace/actions/conan-cache

**Comment by cassava** [comment]
> This is a simple Github Actions workflow that will run a cloe build + smoketest on each push. If we find this is too much, we can also limit it to pull-requests only.

Ok, I've limited it to pull_request only.

File: docs/requirements.txt [hunk 1]
```diff
@@ -0,0 +1,5 @@
+sphinx_rtd_theme
```
[the remark this answers is not in the document]
Comment (cassava) [reply docs/requirements.txt:1]: Agreed.

File: Makefile.setup [hunk 2]
```diff
@@ -99,6 +99,7 @@ install-archlinux-deps::
 		curl \
 		doxygen \
 		git \
+		graphviz \
```
[the remark this answers is not in the document]
Comment (cassava) [reply Makefile.setup:102]: 🤦‍♂️ Oops, you're right!

**Comment by cassava** [comment]
Ok, I addressed your findings and cleaned up the commits.

**Comment by tobifalk** [comment]
LGTM!

**Comment by cassava** [comment]
This PR also fixes #71.

**Pull request merged**